dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -106,9 +106,56 @@ namespace {
 
 void Spdp::init(DDS::DomainId_t /*domain*/,
                        DCPS::RepoId& guid,
-                       const DDS::DomainParticipantQos& /*qos*/,
+                       const DDS::DomainParticipantQos& qos,
                        RtpsDiscovery* disco)
 {
+  bool disable_writers = false;
+
+  const DDS::PropertySeq& properties = qos.property.value;
+  for (unsigned int idx = 0; idx != properties.length(); ++idx) {
+    const char* name = properties[idx].name.in();
+    if (std::strcmp("OpenDDS.Rtps.Discovery.DisableWriters", name) == 0) {
```

Comment:
I'm looking for suggestions as to what header should define the constant "OpenDDS.Rtps.Discovery.DisableWriters", what the actual key should be, and what the value should be and how we should handle it.

Reply:
a. RtpsDiscovery.h
b. "OpenDDS.RtpsDiscovery.EndpointAnnouncements"
c. "0" or "false" (case insensitive) to disable

Reply:
I think there are normally 3 heartbeats without security, right? Publications, Subscriptions, and Participant Messages? And more for security... does "EndpointAnnouncements" cover the rest of them? Also, does this need to be a qos setting? Can it just be be a config entry?

Reply:
We are leaving Participant Message (liveliness) for a future iteration, but I agree it may be an issue.  We can rename the constant when we get there, if needed.

It is QoS because it impacts a specific DomainParticipant.  The config file only works at the domain level.

Reply:
Looking at the spec again, I think we could go ahead and treat `participant_message_writer_` and `participant_message_secure_writer_` the same as the endpoint discovery writers.  There's nothing that could be written on "participant messages" (liveliness updates) that would be relevant in this scenario.

Reply:
> Also, does this need to be a qos setting? Can it just be be a config entry?

We wanted it in the qos so that you can selectively apply it to specific participants.

@mitza-oci Are you still okay with "OpenDDS.RtpsDiscovery.EndpointAnnouncements"?

Reply:
Yes, unless anyone has other suggestions?